src/controller/api/thought.js

```diff
@@ -0,0 +1,150 @@
+const router = require("express").Router();
+
+const { Thought, User } = require("../../models/");
+
+module.exports = {
+  // Get all thoughts
+  async getThoughts(req, res) {
+    try {
+      const thoughts = await Thought.find({});
+      res.json({ thoughts });
+    } catch (error) {
+      console.log(error);
+      res
+        .status(500)
+        .send({ msg: "Something went wrong while retrieving thoughts." });
+    }
+  },
+  // Get single thought
+
+  async getSingleThought(req, res) {
+    try {
+      const thought = await Thought.findOne({ _id: req.params.thoughtId })
+        .select("-__v")
+        .populate({ path: "username", select: "-__v" });
+      if (!thought) {
+        res.status(404).json({ msg: "No thought with this ID." });
+      } else {
+        res.json(thought);
+      }
+    } catch (error) {
+      res
+        .status(500)
+        .send({ msg: "Something went wrong while retrieving the thought." });
+    }
+  },
+
+  // Create thought
+  async createThought(req, res) {
+    try {
+      const newThought = await Thought.create({
+        ...req.body,
+      });
+      // Update user with their new thought
+      const user = await User.findOneAndUpdate(
+        { username: req.body.username },
+        { $addToSet: { thoughts: newThought } },
+        { new: true }
+      );
+      if (!user) {
+        res.status(404).json({ msg: "No user with this ID." });
+      }
+      res.json({ user, msg: "Thought created" });
+    } catch (error) {
+      console.log(error);
+      res
+        .status(500)
+        .json({ msg: "Something went wrong while creating the thought." });
+    }
+  },
+  // Update thought
+  async updateThought(req, res) {
+    try {
+      const thought = await Thought.findOneAndUpdate(
```

Comment:
consider adding some validation logic here too, so that the thought is only updated if the body of the request contains the correct information.